metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/utils/SyslogUtils.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.parsers.utils;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class SyslogUtils {
+
+    public static long convertToEpochMillis(String logTimestamp, String logTimeFormat) {
+        ZonedDateTime timestamp = ZonedDateTime.parse(logTimestamp, DateTimeFormatter.ofPattern(logTimeFormat).withZone(ZoneOffset.UTC));
+        return timestamp.toEpochSecond() * 1000;
+    }
+
+    public static long parseTimestampToEpochMillis(String logTimestamp) {
+        if (logTimestamp.length() < 20) {
+            ZonedDateTime now = ZonedDateTime.now(ZoneOffset.UTC);
+            int year = now.getYear();
+            if (now.getDayOfYear() == 1 && !now.getMonth().toString().substring(0,3).equals(logTimestamp.substring(0,3).toUpperCase()))
+                year--;
```

Comment:
Sorry for the late comment, but I perceive a small problem here.  
1. Using criteria "now == day#1 in UTC" has an unknown relationship to the incoming data, depending on the source time zone.  You might be 24 hrs ahead of the source, or you might be one second ahead.  So if there might be a delay in obtaining the logs, due perhaps to a half-day network failure at Metron's end, you will backdate logs from some sources but not from others, depending on their geographic location.  This argument is not alleviated by changing from "day #1" to "day #1 or #2", the specifics just change somewhat.  Also, log ingest can be delayed for months rather than days, and we want to be able to accommodate replay or historical ingest.

I would suggest instead of using the current criterion 
(now.getDayOfYear() < "configurable.number.of.days" && now.getMonth() != logTimeStamp.getMonth()) 
that we use  the criterion that the logTimestamp calculated with "this year" is significantly in the future from "now".  I suggest "more than 4 days in the future" is a good criterion; that leaves 1 day for NTP error on the logging side, 1 day for NTP error on the Metron side, 1 day for possible leap year artifacts, and 1 day for Murphy's Law.  With this rule, year-less logs can be correctly ingested from up to one year (less 4 days) in the past.

2. For the stated goal, that's sufficient.  We don't need to make the "future limit" configurable.  However, there is also the use case of ingesting year-less data more than a year old.  The only way to enable that, that I can see, would be configuring the source with a year (or an age) along with a timezone.  Doesn't work for on-going sources, but would work for a source that ingests a chunk of history of known age.

3. BTW, is the ZonedDateTime.parse() method robust if we accidentally hand it a FEB 29 date from a non-leap year?  Since we are trying to synthesize the "year" information, it could happen.